src/archive_write_files.cpp

```diff
@@ -48,7 +48,7 @@ const char* const pb_format =
   for (std::string file : files) {
     stat(file.c_str(), &st);
     entry = archive_entry_new();
-#if defined(_WIN32) || (!defined(__GNUC__) && !defined(__clang__))
```

Comment:
I am not sure why we need this PR to be honest. Especially this line, which also changes behavior on non-GNU compilers, not only on Windows.

Reply:
then what about https://github.com/r-lib/archive/pull/80#discussion_r1234284267 ... your stance is different...